src/pages/project/helpers/projectRoleHelpers.ts

```diff
@@ -0,0 +1,102 @@
+import { ProjectContributor, ProjectContributorRole, ProjectContributorType } from '../../../types/project.types';
+
+export const isProjectManagerRole = (role: ProjectContributorRole) => {
+  return role.type === 'ProjectManager';
+};
+
+export const isNonProjectManagerRole = (role: ProjectContributorRole) => {
+  return role.type !== 'ProjectManager';
+};
+
+export const findProjectManagerRoleIndex = (contributor: ProjectContributor) => {
+  return contributor.roles.findIndex((role) => isProjectManagerRole(role));
+};
+
+export const findProjectManagerRole = (contributor: ProjectContributor) => {
+  const projectManagerRoleIndex = findProjectManagerRoleIndex(contributor);
+
+  return projectManagerRoleIndex > -1 ? contributor.roles[projectManagerRoleIndex] : undefined;
+};
+
+export const replaceRolesOnContributor = (
+  contributors: ProjectContributor[],
+  contributorIndex: number,
+  newRoles: ProjectContributorRole[]
+) => {
+  const newContributors = [...contributors];
+  const newContributor = { ...contributors[contributorIndex] };
+  newContributor.roles = newRoles;
+  newContributors[contributorIndex] = newContributor;
+  return newContributors;
+};
+
+export const deleteRoleFromContributor = (
+  contributors: ProjectContributor[],
+  contributorIndex: number,
+  roleIndex: number
+) => {
+  const newRoles = [...contributors[contributorIndex].roles];
+  newRoles.splice(roleIndex, 1);
+  return replaceRolesOnContributor(contributors, contributorIndex, newRoles);
+};
+
+export const deleteProjectManagerRoleFromContributor = (contributors: ProjectContributor[]) => {
+  const projectManagerIndex = contributors.findIndex((c) => c.roles.some((role) => isProjectManagerRole(role)));
+  const projectManager = contributors[projectManagerIndex];
+  const projectManagerRoleIndex = projectManager?.roles.findIndex((role) => isProjectManagerRole(role));
+
+  return projectManagerRoleIndex > -1
+    ? deleteRoleFromContributor(contributors, projectManagerIndex, projectManagerRoleIndex)
+    : contributors;
```

Comment:
Personlig syns jeg det er vanskelig å henge med på alt som skjer når man må hoppe via flere ulike metoder. Virker som vi her egeentlig bare vil fjerne prosjektleder-rollen, som virker rett frem, men der er det selvsagt mulig jeg bare er veldig naiv vedrørende prosjektmodellen også 😅  
```suggestion
  const newRoles = projectManager.roles.filter((role) => !isProjectManagerRole(role));
  contributors[projectManagerIndex] = { ...projectManager, roles: newRoles };
  return contributors;
```

Reply:
Ja noe av det er sikkert overkill, men ofte ble det delt opp slik fordi deler av funksjonen var felles for flere andre funksjoner, og jeg tenker felles kode er bra for å hindre bugs.